solidity-v1/dashboard/src/pages/threshold/ThresholdUpgradePage.jsx

```diff
@@ -0,0 +1,333 @@
+import * as Icons from "../../components/Icons"
+import NavLink from "../../components/NavLink"
+import TokenAmount from "../../components/TokenAmount"
+import { KEEP } from "../../utils/token.utils"
+import AllocationProgressBar from "../../components/threshold/AllocationProgressBar"
+import UpgradeTokensTile from "../../components/threshold/UpgradeTokensTile"
+import resourceTooltipProps from "../../constants/tooltips"
+import useKeepBalanceInfo from "../../hooks/useKeepBalanceInfo"
+import useGrantedBalanceInfo from "../../hooks/useGrantedBalanceInfo"
+import { add, lte } from "../../utils/arithmetics.utils"
+import {
+  useWeb3Address,
+  useWeb3Context,
+} from "../../components/WithWeb3Context"
+import { useDispatch, useSelector } from "react-redux"
+import { useModal } from "../../hooks/useModal"
+import { LINK, MODAL_TYPES } from "../../constants/constants"
+import TokenAmountSkeleton from "../../components/skeletons/TokenAmountSkeleton"
+import ResourceTooltip from "../../components/ResourceTooltip"
+
+const ThresholdUpgradePage = () => {
+  const { isConnected } = useWeb3Context()
+  const address = useWeb3Address()
+  const dispatch = useDispatch()
+  const { openModal } = useModal()
+
+  useEffect(() => {
+    if (isConnected) {
+      dispatch({
+        type: "staking/fetch_delegations_request",
+        payload: { address },
+      })
+      dispatch({
+        type: "token-grant/fetch_grants_request",
+        payload: { address },
+      })
+    }
+  }, [dispatch, isConnected, address])
+
+  const { delegations, undelegations, isDelegationDataFetching } = useSelector(
+    (state) => state.staking
+  )
+
+  const { grants, isFetching: isGrantDataFetching } = useSelector(
+    (state) => state.tokenGrants
+  )
+
+  const { value: keepBalance, isFetching: isKeepBalanceFetching } = useSelector(
+    (state) => state.keepTokenBalance
+  )
+
+  const isDataFetching = useMemo(() => {
+    return (
+      isDelegationDataFetching || isGrantDataFetching || isKeepBalanceFetching
+    )
+  }, [isDelegationDataFetching, isGrantDataFetching, isKeepBalanceFetching])
+
+  const { totalOwnedStakedBalance } = useKeepBalanceInfo()
+
+  const { totalGrantedStakedBalance } = useGrantedBalanceInfo()
+
+  const totalGrantedReadyToReleaseTokens = useMemo(() => {
+    if (isDataFetching || grants.length === 0) return "0"
+    return grants
+      .map(({ readyToRelease }) => readyToRelease)
+      .reduce(add, "0")
+      .toString()
+  }, [grants, isDataFetching])
+
+  const notStakedTotalAmount = useMemo(() => {
+    if (isDataFetching) return "0"
+    return add(keepBalance, totalGrantedReadyToReleaseTokens).toString()
+  }, [keepBalance, totalGrantedReadyToReleaseTokens, isDataFetching])
+
+  // const notStakedTotalAmount = useMemo(() => {
+  //   return add(
+  //     totalOwnedUnstakedBalance,
+  //     totalGrantedUnstakedBalance
+  //   ).toString()
+  // }, [totalOwnedUnstakedBalance, totalGrantedUnstakedBalance])
```

Comment:
Can we remove it?

Reply:
Fixed in 661c6ed7c202611523d18115787275246d8ebe17